.eslintrc.js

```diff
@@ -31,6 +31,19 @@ module.exports = {
     "@typescript-eslint/no-non-null-assertion": 0,
     "@typescript-eslint/no-explicit-any": 0,
     "react/prop-types": 0,
+    "@typescript-eslint/no-restricted-imports": [
+      "error",
+      {
+        paths: [
+          {
+            name: "react-i18next",
+            importNames: ["useTranslation"],
+            message:
+              "Please import useTranslation from '~/hooks/useTranslation' instead.",
+          },
+        ],
+      },
+    ],
```

Comment:
Cool way to handle it. Maybe later we could also use this to prevent importing from deep in the modules folder 🤔 so only allow modules/xxx imports

Reply:
Yes! I only recently learned about it and it's quite helpful for situations like these.

One really nice usecase I sometimes have is also if you have packages installed only for importing types from them, but it would be a big problem if the actual import landed in the bundle somehow (e.g. because of size, or because it has node-only code, or it would create a bad circular dependency or something).

With `@typescript/eslint/no-restricted-imports` you can then forbid importing from that package, but [ set `allowTypeImports: true`](https://typescript-eslint.io/rules/no-restricted-imports/#allowtypeimports), and it won't complain for any `import type`.